### ISSUE TITLE
[Bug]: Inconsistent chunk status for concurrent COMPRESS - INSERT

### ISSUE THREAD

**Issue by konskov** [timescale/timescaledb#5389]
### What type of bug is this?

Incorrect result

### What subsystems and features are affected?

Compression

### What happened?

Adding the following permutation in `compression_ddl_iso`
```
permutation "LockChunk1" "C1" "I1" "UnlockChunk" "Cc" "Ic" "SC1" "S1"
```
Notice that the row inserted by the Insert operation is not returned when querying the table, because its status is (1), when it should be partial (9), due to the new insert. 

### TimescaleDB version affected

2.11.0-dev

### PostgreSQL version used

13.6

### What operating system did you use?

Macos ventura 13.1

### What installation method did you use?

Source

### What platform did you run on?

Not applicable

### How can we reproduce the bug?

```bash
Following the steps of the compression_ddl_iso test
```

**Comment by konskov** [comment]
This does not affect any of our released versions. I only created this issue for tracking

**Comment by konskov** [comment]
Updating the issue with possible solutions suggested so far: 
| Idea/solution      | Branch for testing | Outcome |
| ----------- | ----------- | ----- |
| Read chunk entry in chunk_dispatch without taking locks on the catalog table. After reading, take a tuple-level lock on the corresponding CHUNK table row: FOR UPDATE lock when the chunk status is compressed but not partial, FOR SHARE lock when chunk status is compressed AND partial   | [Title](https://github.com/konskov/timescaledb/tree/compression_locks_chunk_status_1)       | inconsistency still occurs |

**Comment by antekresic** [comment]
Fixed by #5533
